src/functions/createReport.ts

```diff
@@ -58,7 +59,10 @@ export function createReport(
         }
         finalArray.push(outputArray[index]);
 
-        if (index == outputArray.length - 1 && !isRestaurantBankrupt) {
+        if (index == outputArray.length - 1 && !isRestaurantBankrupt && totalTrashTaxValue == 0) {
+            finalArray.push(`Restaurant budget: ${restaurantBudgetIterations[restaurantBudgetIterations.length - 1]}`);
+        } else if(index == outputArray.length - 1 && !isRestaurantBankrupt && totalTrashTaxValue != 0){
+            finalArray.push(`We paid: ${totalTrashTaxValue} for exceeded waste.`);
```

Comment:
Step 1:
`(index == outputArray.length - 1` <- maybe it should be extracted to named var.
Step 2: realization:
maybe.. move this logic BELOW `for` so you don't have to check index.
Less code, less errors, less computing power needed